src/sentry/models/group.py

```diff
@@ -1223,14 +1223,20 @@ def update_group_open_period(
         return
 
     if new_status == GroupStatus.RESOLVED:
+        if activity is None:
+            activity = (
+                Activity.objects.filter(
+                    group=group,
+                    type=ActivityType.SET_RESOLVED.value,
+                )
+                .order_by("-datetime")
+                .first()
+            )
+
         open_period.update(
-            date_ended=group.resolved_at if group.resolved_at else timezone.now(),
+            date_ended=group.resolved_at if group.resolved_at else activity.datetime,
             resolution_activity=activity,
             user_id=activity.user_id if activity else None,
         )
-    elif (
-        new_status == GroupStatus.UNRESOLVED
-        and group.substatus != GroupSubStatus.REGRESSED
-        and should_reopen_open_period
-    ):
+    elif new_status == GroupStatus.UNRESOLVED and should_reopen_open_period:
```

Comment:
I initially had the substatus exclude regressions here to avoid duplicate logic for errors. That's actually overly cautious and not really necessary. The only way we'd get to this point for a regression is for non-error issues or for regressions handled via integrations. Both of those cases should update the open period too.